x-pack/plugins/apm/ftr_e2e/README.md

```diff
@@ -2,6 +2,144 @@
+
+### Setup intercepts before opening the page
+
+It is important that interceptors are setup before opening the page that fires the requests that are intercepted. If the interceptors are setup after the requests were made, they will not be captured and the test will timeout during `cy.wait`,
+
+**WRONG**
+
+```ts
+it('calls the dependencies API', () => {
+  cy.visit('/app/apm/services');
+  cy.intercept('GET', '/internal/apm/dependencies/top').as('topDependencies');
+  cy.wait('@topDependencies');
+});
+```
+
+**Correct**
+
+```ts
+it('calls the dependencies API', () => {
+  cy.intercept('GET', '/internal/apm/dependencies/top').as('topDependencies');
+  cy.visit('/app/apm/services');
+  cy.wait('@topDependencies');
+});
+```
+
+### Prefer `cy.visitKibana` instead of `cy.visit`
+
+In most cases we should use [`cy.visitKibana`](https://github.com/elastic/kibana/blob/50821db39c07d5d35d510c8082d5c608c4e2fd4e/x-pack/plugins/apm/ftr_e2e/cypress/support/commands.ts#L51-L56) instead of `cy.visit`.
+`cy.visitKibana` will wait for Kibana to have successfully loaded before moving on. This will reduce the risk of timing out later in the test because we split up the wait time in two parts: Kibana load time, and APM load time thus a time budget for each (by default 40 seconds).
+
+### Clean data before and after each test
+
+Some times test can stop in the middle of the execution and start running again, making sure that, if there were some data created, is properly cleaned before starting the test again will guarantee the proper execution of the test.
+
+**WRONG**
+
+```ts
+// we check that there are not links created
+it('shows empty message and create button', () => {
+  cy.visitKibana(basePath);
+  cy.contains('No links found');
+  cy.contains('Create custom link');
+});
+
+it('creates custom link', () => {
+  cy.contains('Create custom link').click();
+  cy.get('input[name="label"]').type('foo');
+  cy.contains('Save').click();
+  cy.contains('foo');
+  // if the test stops before the delete and starts again, the previous test will fail
+  cy.contains('Delete').click();
+});
+```
+
+**CORRECT**
+
+Delete any possible data created during the test before running them, preferably calling the api
+
+```ts
+describe('Custom links', () => {
+  beforeEach(() => {
+    cy.request({
+      log: false,
+      method: 'DELETE',
+      url: `${kibanaUrl}/internal/apm/settings/custom_links/link.id`,
+      body: {},
+      headers: {
+        'kbn-xsrf': 'e2e_test',
+      },
+      auth: { user: 'editor', pass: '****' },
+    });
+  });
+});
+```
+
+Use `synthtrace.clean()` after each test suit
+
+```ts
+describe('when data is loaded', () => {
+    before(() => {
+      synthtrace.index(
+        generateData({
+          from: new Date(start).getTime(),
+          to: new Date(end).getTime(),
+        })
+      );
+    });
+
+    after(() => {
+      synthtrace.clean();
+    });
+
+    it(...)
+});
+```
```

Comment:
I think you should use the same example as above, instead of introducing a new example

```suggestion
```ts
beforeEach(() => {
 // deleteAllCustomLinks();
})

it('shows empty message and create button', () => {
  // ...
});
it('creates custom link', () => {
  // ...
});
```
```

Reply:
Thanks for the suggestions @sqren 🎉